internal/testing/cluster.go

```diff
@@ -92,6 +92,21 @@ type TestCluster struct {
 	wg     sync.WaitGroup
 }
 
+func (tc *TestCluster) ShutdownAutopilot(ctx context.Context) error {
+	stopFn1 := tc.cleanups[len(tc.cleanups)-2]
```

Comment:
This is very fragile and hard to read. I'd rather split up `cleanups` into multiple fields. 
e.g. `busCleanup`, `workerCleanups`, `autopilotCleanup` etc.

Reply:
ok, I agree I just figured the cleanup functions were unlikely to be changed